### ISSUE TITLE
IF ELSE IF chains can (and should) also be implemented for code export to other languages (like the C family)

### ISSUE THREAD

**Issue by codemanyak** [fesch/Structorizer.Desktop#1148]
OBERON export Bug was fixed:
![grafik](https://github.com/fesch/Structorizer.Desktop/assets/15326471/8f60ef5a-74e6-47d8-ab26-431578e5a516)

_Originally posted by @codemanyak in https://github.com/fesch/Structorizer.Desktop/issues/1146#issuecomment-2005219412_
            
This improvement should also be applied to other languages which support it somehow (though not with a special keyword), e. g. C, C++, C#, Java, etc.:
![grafik](https://github.com/fesch/Structorizer.Desktop/assets/15326471/6295b4d4-ff33-4309-96f3-5d9a9c687c51)

**Comment by codemanyak** [comment]
Implemented for version 3.32-19

**Comment by codemanyak** [comment]
Implementation of optimised code export for IF ELSE IF chains for Pascal added (ready for version 3.32-20).

**Comment by codemanyak** [comment]
@GitMensch 
> I do wonder if the code export for COBOL should use:

Hi Simon! It looks plausible and optically close to IF cond1 ... ELSE IF cond2 ..,, albeit somewhat ugly (for a non-COBOL-afficionado). And it's not that straightforward to implement as for other languages where a simple loop can be used to insert as many ELSIF clauses as the nesting of Alternatives allows. Here we must do a fundamental decision between nested IF and EVALUATE in advance. It is of course feasible but not as elegant as for "usual" programming languages. Maybe, an export option should be introduced for COBOL to decide whether thi translation is to be preferred where possible.

**Comment by GitMensch** [comment]
@codemanyak 

An output setting makes this even more complicated (for not much benefit), no? So I'd suggest to not add the option.
But I do see the point of "needs a decision up front". Maybe put that on the backlist then...

**Comment by codemanyak** [comment]
I prepared the implementation in the draft COBOL generator with an else-if threshold of 2 now:
![grafik](https://github.com/fesch/Structorizer.Desktop/assets/15326471/00f3798c-dddd-4147-9d65-0ee45de17501)

**Comment by GitMensch** [comment]
That looks good and can be un-drafted.
I just wonder what happened to the export, I've thought we had simple assignments (`MOVE var/lit TO var`) and possibly user-input data (`ACCEPT var`) implemented back then.

... but maybe that was just the import.

**Comment by codemanyak** [comment]
@GitMensch 
> [...] I just wonder what happened to the export, I've thought we had simple assignments (`MOVE var/lit TO var`) and possibly user input/output data (`ACCEPT var`, `DISPLAY var`) implemented back then.
> ... but maybe that was just the import.

Yes, itwas just the import. The export still has some substantial deficiencies.I became aware of on this occasion. The decomposition of general (mixed-type) expressions in instructions still remains a problem. I have to push forward the #800 redesign because this appears to be the clue for a sensible analysis.
 
> [Note: I _guess_ the `CODE SECTION.` is intended to be output directly before the first instruction, not after every one.]

I will have a look at it.

**Comment by codemanyak** [comment]
The if-else-if chain optimisation has been forgotten in the shell generators.

**Comment by codemanyak** [comment]
If-elif-else-fi now also realised fr bash and ksh. Fix available for next version (3.32-33).
